### ISSUE TITLE
analyzer code presubmit checks

### ISSUE THREAD

**Issue by srawlins** [dart-lang/sdk#53578]
We brought up recently checks that we might want to make presubmit (on upload) checks for analyzer, Hopefully we can make some of these only trigger when certain files are changed:

* any pkg/{analyzer,analysis_server,linter,...} changes: check that files are sorted.
* touching pkg/analyzer/messages.yaml: run `pkg/analyzer/tool/messages/messages_test.dart`
* touching pkg/analyzer/messages.yaml: check that Dart error code has been generated (`dart pkg/analyzer/tool/messages/generate.dart`)
* touching `pkg/{analyzer/analysis_server}/test` files: check that `test_all.dart` files are updated.

CC @bwilkerson @scheglov @kallentu @pq @keertip

**Comment by pq** [comment]
Emphatic 💯💯💯. 

My understanding is that presubmits are largely written in python but hopefully we can (performantly) call out to `dart` so we don't have to duplicate logic we've already got?

@whesse, @athomas: maybe you could advise?

**Comment by bwilkerson** [comment]
We do need to be careful not to add to much time to the presubmits. We might want to time some of these tests to ensure that they're all reasonable. Of course, it won't be as big an issue if they can run only when a specific file has been modified.

**Comment by scheglov** [comment]
I agree that any additional checks that we add should be fast, or we don't add them.
I already manually comment out a few, such as
```
    // verify_diagnostics.main();
    // verify_docs.main();
    // verify_tests.main();
```

and `pkg/analysis_server/test/benchmarks_test.dart`

Because they take too much time, and I care about these checks only when I'm ready to sent CL.
No while I'm doing some work on element model that might affect many test, so I often run all of them.

Sometimes this causes me an issue that something fails on bots because I forgot to check.
But more often it would not, and I will just waste time running these checks.

**Comment by srawlins** [comment]
On my laptop, `verify_sorted_test.dart` takes 20 seconds, which validates that files in all of our packages are sorted. Breaking into smaller tests, like just validating `pkg/analyzer` cuts down the time. `pkg/analyzer` takes 9 seconds, `pkg/analysis_server` takes 9 seconds.

`messages_test.dart` takes 2 seconds.
`verify_tests_test.dart` takes 3 seconds.
`verify_error_fix_status_test.dart` takes ~0 seconds.
